modelling/src/main/java/org/axonframework/modelling/annotations/InjectEntityParameterResolver.java

```diff
@@ -74,24 +77,31 @@ public InjectEntityParameterResolver(
     @Nullable
     @Override
     public Object resolveParameterValue(@Nonnull ProcessingContext context) {
+        return FutureUtils.joinAndUnwrap(resolveParameterValueAsync(context));
+    }
+
+    @Nonnull
+    @Override
+    public CompletableFuture<Object> resolveParameterValueAsync(@Nonnull ProcessingContext context) {
         Message message = Message.fromContext(context);
 
         try {
             Object resolvedId = identifierResolver.resolve(message, context);
             StateManager stateManager = configuration.getComponent(StateManager.class);
+
             if (managedEntity) {
-                return stateManager.loadManagedEntity(type, resolvedId, context).join();
+                return stateManager.loadManagedEntity(type, resolvedId, context)
+                                  .thenApply(entity -> entity);
```

Comment:
Is the `thenApply` doing anything useful here?  If so, perhaps add a comment as it confuses me.  Same for the one below.

Reply:
I've changed it to casting.